lib/http_aws_sigv4.c

```diff
@@ -250,6 +251,17 @@ CURLcode Curl_output_aws_sigv4(struct Curl_easy *data, bool proxy)
   memcpy(date, timestamp, sizeof(date));
   date[sizeof(date) - 1] = 0;
 
+  if(data->set.postfieldsize < 0)
+    post_data_len = strlen(post_data);
+  else
+    post_data_len = (size_t)data->set.postfieldsize;
```

Comment:
This typecast is wrong, it is a `curl_off_t`. I see now that you didn't add this but only moved the line, but the type should still be fixed.

Reply:
@bagder Should the def on line 95, `size_t post_data_len` also be changed to `curl_off_t post_data_len`?